src/main/java/org/apache/commons/lang3/RandomStringUtils.java

```diff
@@ -293,7 +297,8 @@ public static String random(final int count, final boolean letters, final boolea
      *  numeric characters
      * @return the random string
      */
-    public static String random(final int count, final int start, final int end, final boolean letters, final boolean numbers) {
+    @SuppressWarnings("index:argument.type.incompatible") // null array is handled differently
```

Comment:
I don't understand what the null array has to do with a warning suppression here. Can you explain this a little more?

Reply:
If the array passed is null, then the set of all chars is used.
The error is issued because if `null` is passed as the array parameter, the other two parameters which are supposed to be `@IndexFor` and `@IndexOrHigh` are not such.
How do you propose I tackle this? Should I suppress the warnings in the `random()` function that is annotated and remove the annotations?

Reply:
I see: the joy of not having optional parameters.

I think you've annotated it correctly, but I would move this text:

> If the array passed is null, then the set of all chars is used.
> The error is issued because if null is passed as the array parameter, the other two parameters which are supposed to be @IndexFor and @IndexOrHigh are not such

into the comment on the code